daemon/daemon.c

```diff
@@ -406,6 +406,7 @@ static bool parse_args(int argc, char const** argv, tr_variant* settings, bool*
 
         case 'd':
             *dump_settings = true;
+            *exit_code = 0;
```

Comment:
The issue itself is identified correctly, but I'd rather follow the other two cases in this function and only set `*exit_code` when `false` is returned. Doing `int ret = 0;` in `tr_main` seems more suitable.